### ISSUE TITLE
The applicability of this software

### ISSUE THREAD

**Issue by MaryGoAround** [immune-health/antigen.garnish#133]
Hello

I have a set of vcf files (SNVs) from patients who have responded to chemotherapy versus a set of patients who have not responded.

I want to quantify the neoantigen load in the tumors ,can I still use your software?

In a paper I read: `We first collected all peptides defined by a 17 amino-acid region centered on the amino acid which changes upon the mutation. We identified mutant nonamers with ≤500 nM binding affinity for patient-specific class I human lymphocyte antigen (HLA) alleles, constituting potential candidate neoantigens. Binding affinities were predicted using NetMHC-3.4 [82]. We then quantified the peptides that displayed high affinity binding in tumor, but low or no binding in the respective matched normal and obtained total counts `

Is this almost the same which your software can do?

Thank you if you guid me if I am on the right track

**Comment by leeprichman** [comment]
Yes, as long as you have properly formatted VCFs and per sample HLA calls, antigen.garnish can be used for this kind of analysis!

**Comment by MaryGoAround** [comment]
Sorry, does your software need HLA type of a given patient (.VCF) ?

I only have access to .vcf (SNVs and INDELs separately) coming from whole genome sequencing

Is this adequate for your software to predict Neoantigen load?

Thank you

**Comment by leeprichman** [comment]
antigen.garnish does not predict HLA alleles. You must know your alleles of interest for each VCF/patient in order to perform neoantigen analysis. 

VCF files only contain lists of variants. They do not contain enough information to determine HLA haplotypes. You will need to use the sequencing data to determine HLA types per patient prior to predicting neoantigen burden with any pipeline.

**Comment by MaryGoAround** [comment]
Sorry in which of the code, your software demands  alleles of interest for each VCF/patient? Can not I use all of human alleles instead? because in this part of your script says

```
# MHC may also be set to "all_human" or "all_mouse" to use all supported alleles

dt[, MHC := c("HLA-A*01:47 HLA-A*02:01 HLA-DRB1*14:67")]
```
Thank you for any help

**Comment by leeprichman** [comment]
You could run all possible MHC alleles but it would be computationally quite intensive and wouldn’t really give you neoantigen burden. Neoantigen burden is determined by the predicted peptides derived from tumor mutations that have affinity for the subjects specific MHC alleles.

**Comment by MaryGoAround** [comment]
Actually, from this paper https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5957269/ I noticed they are using this approach just using these HLAs

```
HLA-A01:01
HLA-A01:04
HLA-A02:01
HLA-A02:06
HLA-A02:89
HLA-A03:01
HLA-A03:21
HLA-A03:26
HLA-A11:01
HLA-A23:01
HLA-A24:02
HLA-A24:03
HLA-A25:01
HLA-A26:01
HLA-A29:01
HLA-A29:02
HLA-A30:01
HLA-A30:02
HLA-A31:01
HLA-A32:01
HLA-A33:01
HLA-A68:01
HLA-B07:02
HLA-B07:05
HLA-B08:01
HLA-B13:02
HLA-B14:02
HLA-B18:01
HLA-B27:05
HLA-B35:01
HLA-B35:02
HLA-B35:12
HLA-B35:40
HLA-B37:01
HLA-B38:01
HLA-B40:02
HLA-B400:102
HLA-B41:01
HLA-B44:02
HLA-B44:03
HLA-B450:101
HLA-B48:01
HLA-B49:01
HLA-B51:01
HLA-B51:08
HLA-B55:01
HLA-B56:01
HLA-B57:01
HLA-C01:02
HLA-C02:02
HLA-C03:03
HLA-C03:04
HLA-C04:01
HLA-C05:01
HLA-C06:02
HLA-C07:01
HLA-C07:02
HLA-C07:04
HLA-C08:02
HLA-C08:03
HLA-C12:03
HLA-C15:02
HLA-C15:04
HLA-C15:05
HLA-C16:01
HLA-C16:02
HLA-C17:01

```

I emailed the author, she says that

**I don’t remember the details anymore, but the HLA types are definitely not predicted from the cohort. I can see if I can find this information, but netMHC documentation should explain how to get this information. You can probably use the same file, but do a bit of a search though in case there have been some updates on HLA types in human in the past 4 years.**

**Comment by leeprichman** [comment]
If you have vcf files, you can run antigen.garnish with whichever MHC alleles you are interested in.

**Comment by beginner984** [comment]
Sorry, please can you suggest/refer me to a `common HLA types in Human` for predicting Neoantigen load as I don't have access to HLA typing specific to my patients

Thank you so much for any help

**Comment by leeprichman** [comment]
I don’t have a suggestion for this. I’m sorry. I don’t know if any resource with a consensus for common alleles

**Comment by beginner984** [comment]
Sorry is your `all_alleles.txt` a combination of `mhcflurry_alleles.txt , netMHCII_alleles.txt , netMHCIIpan_alleles.txt , netMHC_alleles.txt and netMHCpan_alleles.txt` ?

**Comment by leeprichman** [comment]
Yes.
